CMakeLists.txt

```diff
@@ -16,7 +16,16 @@ set(VM_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src/vm)
 
 if(CMAKE_SYSTEM_NAME STREQUAL Linux)
     set(CLR_CMAKE_PLATFORM_UNIX 1)
-    set(CLR_CMAKE_PLATFORM_UNIX_TARGET_AMD64 1)
+    if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64)
+        set(CLR_CMAKE_PLATFORM_UNIX_TARGET_AMD64 1)
+    elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l)
+        set(CLR_CMAKE_PLATFORM_UNIX_TARGET_ARM 1)
+        # Because we don't use CMAKE_C_COMPILER/CMAKE_CXX_COMPILER to use clang
+        # we have to set the triple by adding a compiler argument
+        add_compile_options(-target armv7-linux-gnueabihf)
+        add_compile_options(-mthumb)
+        add_compile_options(-mfpu=vfpv3)
```

Comment:
They are the assumptions made on windows, but we can compile time confirm them on !windows.